PcapDotNet/src/PcapDotNet.Packets/Dhcp/DhcpDatagram.cs

```diff
@@ -0,0 +1,368 @@
+using PcapDotNet.Packets.Dhcp.Options;
+using PcapDotNet.Packets.Ethernet;
+using PcapDotNet.Packets.IpV4;
+
+namespace PcapDotNet.Packets.Dhcp
+{
+    /// <summary>
+    /// RFC 2131.
+    /// The DHCP message structure is shown below:
+    /// <pre>
+    /// 0                   1                   2                   3
+    /// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |     op (1)    |   htype (1)   |   hlen (1)    |   hops (1)    |
+    /// +---------------+---------------+---------------+---------------+
+    /// |                            xid (4)                            |
+    /// +-------------------------------+-------------------------------+
+    /// |           secs (2)            |           flags (2)           |
+    /// +-------------------------------+-------------------------------+
+    /// |                          ciaddr  (4)                          |
+    /// +---------------------------------------------------------------+
+    /// |                          yiaddr  (4)                          |
+    /// +---------------------------------------------------------------+
+    /// |                          siaddr  (4)                          |
+    /// +---------------------------------------------------------------+
+    /// |                          giaddr  (4)                          |
+    /// +---------------------------------------------------------------+
+    /// |                                                               |
+    /// |                          chaddr  (16)                         |
+    /// |                                                               |
+    /// |                                                               |
+    /// +---------------------------------------------------------------+
+    /// |                                                               |
+    /// |                          sname   (64)                         |
+    /// +---------------------------------------------------------------+
+    /// |                                                               |
+    /// |                          file    (128)                        |
+    /// +---------------------------------------------------------------+
+    /// |                                                               |
+    /// |                          options (variable)                   |
+    /// +---------------------------------------------------------------+
+    /// </pre>
+    /// </summary>
+    public class DhcpDatagram : Datagram
+    {
+        private static class Offset
+        {
+            public const int Op = 0;
+            public const int Htype = 1;
+            public const int Hlen = 2;
+            public const int Hops = 3;
+            public const int Xid = 4;
+            public const int Secs = 8;
+            public const int Flags = 10;
```

Comment:
Thanks, but I meant split it to different fields:
I think it would make the usage easier.
```
/// +-------------------------------+------------+------------------+
/// |           secs (2)            |  Broadcase | reserved         |
/// +-------------------------------+------------+------------------+
```